### DISCUSSION TITLE
Kiosk Apps with WindowsAppSDK / WinUI3

### DISCUSSION THREAD

**Discussion by myokeeh** [microsoft/WindowsAppSDK#2653] [Q&A]
@andrewleader, what is the latest on running WindowsAppSDK / WinUI3 apps as kiosk apps? I'm in the middle of converting a UWP kiosk app only to discover this is still unsupported. I've added my thoughts on product board some time ago and provided an update recently.

**Comment by myokeeh** [comment]
@andrewleader Following up.

**Reply by andrewleader** [reply]
Hi @myokeeh, I no longer work on WinAppSDK, but I've asked the team to follow up on this. Thanks!

**Comment by BRB2000** [comment]
I am also in desperate need for this feature. Another question would be, why the product board hasn't been updated since the launch of Windows App SDK version 1.1.

**Comment by BenJKuhn** [comment]
I discussed this at some length with the team that owns this area. The short answer is that at least for now kiosk mode is limited to UWP applications, and a change to that will require changes to the OS itself since there are some security model considerations.

Please consider opening this ask in feedback hub so it's tracked as a platform request. While the team is now aware of the gap, direct customer input is always appreciated.

**Reply by myokeeh** [reply]
If I go by my luck with feedback like this, I'd still be waiting on a fix in 5-10 years. I'm literally still waiting on a fix for a bug reported in 2017.

Regardless, I filed it: https://aka.ms/AAhs28d

**Reply by myokeeh** [reply]
Also, I do like the security provided by UWP and I just want my app to function the same way but using WinUI3 / WindowsAppSDK (no new features). I'm basically 99% done except for this Kiosk app hurdle.

**Reply by BRB2000** [reply]
@BenJKuhn The feedback [https://aka.ms/AAhs28d](https://aka.ms/AAhs28d) was added to a collection with the title "Please ensure feedback filed has a detailed title and description". I think that the issue was described in enough detail (the feedback included this GitHub discussion). I suppose that means that it will not be processed any further. What can we do next to bring this issue to attention?

**Reply by myokeeh** [reply]
> @BenJKuhn The feedback [https://aka.ms/AAhs28d](https://aka.ms/AAhs28d) was added to a collection with the title "Please ensure feedback filed has a detailed title and description". I think that the issue was described in enough detail (the feedback included this GitHub discussion). I suppose that means that it will not be processed any further. What can we do next to bring this issue to attention?

That's what that was. I couldn't figure out what feedback got put in the collection.

**Reply by BenJKuhn** [reply]
I've forwarded the feedback hub issue to the owning team and asked them to re-triage. Also, @codendone is going to take over tracking this from here.

**Reply by myokeeh** [reply]
@codendone do you happen to have anything new from the team?

**Reply by BenJKuhn** [reply]
I wouldn't hold my breath for something to happen fast here. This is going to be a deeper change in Windows, and while the team is discussing options, this is most likely something that would be considered for a future OS update. It's not within the scope of things that can be easily addressed in App SDK. If you need to use kiosk mode for an app sooner, you're probably best off continuing with UWP for the time being.

Ben

**Reply by myokeeh** [reply]
Incredibly frustrating.

**Reply by BRB2000** [reply]
@BenJKuhn Thank you very much for bringing this to the attention of the responsible team. I really appreciate it.

**Reply by BenJKuhn** [reply]
I sympathize. 


I did some more digging. People have been using Windows for kiosks long before UWP existed, so I tried to find how it's been done for desktop apps in the past. 

This article provides a summary of many ways to configure a kiosk, including alternate ways to configure the assigned access feature, and may be worth exploring more. There were two that looked particularly interesting.
https://docs.microsoft.com/en-us/windows/configuration/kiosk-methods

1) This article has more information about how to configure a desktop app for use as a kiosk, but only for enterprise and education SKUs. I'm not sure if that's helpful for your situation or not, but I hope so:
https://docs.microsoft.com/en-us/windows/configuration/kiosk-single-app#wizard

2) This article seems to suggest that with PowerShell you might be able to also set up a kiosk on pro SKUs using desktop apps. It's not entirely clear from the way it's written:
https://docs.microsoft.com/en-us/windows/configuration/kiosk-mdm-bridge

I won't be able to dig in more for now, but if any of those options are helpful, please let us know.

Ben

**Reply by myokeeh** [reply]
@codendone, can you provide an update from the owning team?

**Reply by BenHarnett** [reply]
Yes please. As UWP doesn't supporty .NET 5 or 6 or 7 we are stuck still writing referenced libraries in .NET Standard 2.0!

**Reply by myokeeh** [reply]
FYI, with Jen Gentleman's help, the feedback was separated out from the other collection: https://aka.ms/AAju3oc Please upvote

**Reply by myokeeh** [reply]
@BenJKuhn @codendone do you have any updates on this?

**Reply by myokeeh** [reply]
Following up on this. Anyone?

**Reply by thirstyape** [reply]
I was rather surprised this morning when I discovered that MAUI apps (WinUI3) cannot be used in single-app kiosk mode. As UWP is effectively dead, its successor should have the same feature set implemented. The ability to develop apps using the current frameworks that can be used as a single-app kiosk is quite important. Please consider increasing the priority of this issue.

Thank you.

**Reply by christiannagel** [reply]
What's the current status with packaged WinUI apps?

https://github.com/microsoft/microsoft-ui-xaml/issues/9828

Using the *Windows Configuration Designer*, an MSIX package can be selected. However, it detects a *Windows Store app*.

How would it be possible to use an *app installer* for WinUI in Kiosk mode?

We need the packaged version for auto update of the application. If packaged WinUI apps cannot be used, how would it be possible to do automatic updates of the application?

**Reply by DarkNoir** [reply]
I'd settle for "single-app kiosk mode in anything actually supported", so I am looking at Blazor WebAssembly PWA in Edge for now.

**Reply by exedor** [reply]
> I'd settle for "single-app kiosk mode in anything actually supported", so I am looking at Blazor WebAssembly PWA in Edge for now.

Ouch

**Comment by exedor** [comment]
I've been caught in the same quagmire. This is a BIG problem. NOT COOL, Microsoft! I get that you have new leadership and lots of new ways of doing things, and that's awesome and has been helpful. For a long time, one of the greatest value propositions with Microsoft technologies was backward compatibility.

Having both developed and managed the development of front-end and back-end tech, I don't appreciate the front-end space much. The timeframes involved in stuff developed, used, adopted, and abandoned in that chaotic, yet creative space, is far too short for many businesses, and even some developers who may not learn as quickly as others, which causes problems for hiring managers. So much time is spent on busy work dealing with the latest severe security flaw, or complete abandonment in the interwoven web of hundreds of packages and dependencies upon them in any app of significant size, is a nightmare. That same nightmare is slowly creeping into all these core technologies that have historically had stability, backward compatibility, and interoperability.

A LOT of the stability that was there is getting lost. I LOVE the less combative, more open source, and collaborative approach Microsoft has taken in the past decade and a half. I'm just hoping that can be balanced in a way that also doesn't cause loss of the longest standing, and IMHO greatest advantages of using Microsoft technologies.

**Reply by DarkNoir** [reply]
Things are definitely degrading, some of these 'open source' projects are left completely unattended. Pull requests are ignored and left hanging. Visual Studio is now being treated like a side project to Visual Studio Code. I will not be renewing my VS Enterprise subscription due to issues that are going unresolved. This issue here is a great example of 'forward' progress leaving behind important features with no real plan to resolve it. Getting tired of being abandoned. It feels like MS is dropping as much as they add nowadays and teams are disconnected from each other.

**Reply by exedor** [reply]
Yeah, I get it. I'm feeling your pain. At the same time, it may be temporary. I was surprised to learn MS has been making investments at the hardware level with quantum computing. There is synergy there. QC processors are not the end-all be-all of hardware, but they definitely have their place and MS seems to be interested in becoming a hybrid Micro Hard/Soft which is interesting.

I have had some success with Visual Studio bug fixing. There are many features in Visual Studio MS Code does not have. Bug reporting and the way that happens is one of them. I need to ask, did you report your problem with Visual Studio? I had a clear problem and it took me less than 3 minutes, using Visual Studio's built-in recording mechanism to demonstrate and show the problem and report it MS. Then it was fixed in the next update.

I've used VS Code. It's OK, but not great. My prior experience leads me to believe there may be a failure to effectively communicate your concerns with VS.

That said, MS has definitely lost appeal by attempting to "be right" in how it serves customers and how it has lost understanding of it's audiences.

**Reply by DarkNoir** [reply]
Yeah, reported it. The reproduction steps are activate document groups and eventually Intellisense fails and error checking fail until you remove the document group (Split screen showing two document wells) . It happens anywhere from 5 to 20 mins of use.  Not very video friendly, but steps are easy enough.  Having used Microsoft tech for more than 4 decades, the pain adds up sometimes.  Apparently not enough people use windows in kiosk mode to make it worth the effort to keep the feature alive.  Since i am pushed toward the browser anyway, no reason to use windows for it, I suppose. A raspberry pi will work for a kiosk.  

VSCode is no Visual Studio for sure. But seems like all documentation and even their engineers push examples for VSCode. Visual Studio users are like the red headed stepchild :P
